README.md

```diff
@@ -64,17 +64,30 @@ class MyDataSource(panoply.DataSource):
 
 The `DataSource` base class exposes the following methods:
 
-###### __init__(self, source, options)
+###### __init__(self, source, options, events)
 
-Constructor. Receives a dictionary with the data `source` parameters (see below) and a dictionary with any additional `options`. Generally, it's safe to disregard the options, however it may be used for performance optimizations, as it contains hints about incremental keys, excluded fields, etc.
+Constructor. Receives a dictionary with the data `source` parameters (see below), a dictionary with any additional `options` and a dictionary with `events` subscriptions. Generally, it's safe to disregard the options, however it may be used for performance optimizations, as it contains hints about incremental keys, excluded fields, etc, and it may contain additional parameters that can't be transferred with the source
```

Comment:
So what you actually want is to have the ability to change the source when invoking a dynamic param function right?
First I am not sure it's a good idea. The API of such dynamic function is pretty simple you get the source and return the parameter values, now allowing for such simple function to change the source is a bit dangerous. try to see if we can prevent it. (even if it means that we will regenerate the access token)
Second, if we can't prevent it then the implementation should be much less cryptic,
instead of passing a map of events to this lambda, which in its turn just creating a stream and passing forward the *args, so now the stream can emit events ( But this ability hardly depends on the fact that it was created with a map of events which is not always the case) you can change this lambda signature to accept the source and an event emitter, do whatever you want and use the emitter to emit the events you want to emit from this lambda